packages/camera/camera_android_camerax/lib/src/camera2_camera_control.dart

```diff
@@ -0,0 +1,135 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:meta/meta.dart' show immutable;
+
+import 'android_camera_camerax_flutter_api_impls.dart';
+import 'camera_control.dart';
+import 'camerax_library.g.dart';
+import 'capture_request_options.dart';
+import 'instance_manager.dart';
+import 'java_object.dart';
+import 'system_services.dart';
+
+/// Class that provides ability to interoperate with android.hardware.camera2
+/// APIs and apply options to its specific controls like capture request
+/// options.
+///
+/// See https://developer.android.com/reference/androidx/camera/camera2/interop/Camera2CameraControl#from(androidx.camera.core.CameraControl).
+@immutable
+class Camera2CameraControl extends JavaObject {
+  /// Creates a [Camera2CameraControl].
+  Camera2CameraControl(
+      {BinaryMessenger? binaryMessenger,
+      InstanceManager? instanceManager,
+      required this.cameraControl})
+      : super.detached(
+            binaryMessenger: binaryMessenger,
+            instanceManager: instanceManager) {
+    _api = _Camera2CameraControlHostApiImpl(
+        binaryMessenger: binaryMessenger, instanceManager: instanceManager);
+    AndroidCameraXCameraFlutterApis.instance.ensureSetUp();
+    _api.createFromInstances(this, cameraControl);
+  }
+
+  /// Constructs an [Camera2CameraControl] that is not automatically attached to a
+  /// native object.
+  Camera2CameraControl.detached(
+      {BinaryMessenger? binaryMessenger,
+      InstanceManager? instanceManager,
+      required this.cameraControl})
+      : super.detached(
+            binaryMessenger: binaryMessenger,
+            instanceManager: instanceManager) {
+    _api = _Camera2CameraControlHostApiImpl(
+        binaryMessenger: binaryMessenger, instanceManager: instanceManager);
+    AndroidCameraXCameraFlutterApis.instance.ensureSetUp();
+  }
+
+  late final _Camera2CameraControlHostApiImpl _api;
+
+  /// The [CameraControl] info that this instance is based on.
+  ///
+  /// Note that options specified with this [Camera2CameraControl] instance will
+  /// have higher priority than [cameraControl].
+  final CameraControl cameraControl;
+
+  /// Updates capture session with options that the specified
+  /// [CaptureRequestOptions] contains.
+  ///
+  /// Options will be merged with existing options, and if conflicting with what
+  /// was previously set, these options will override those pre-existing. Once
+  /// merged, these values will be submitted with every repeating and single
+  /// capture requests issued by CameraX.
+  Future<void> addCaptureRequestOptions(
```

Comment:
I am not 100% sure that we don't need it, but I was just going to wrap it later if I do.

My understanding, though, is that using [`clearCaptureRequestOption`](https://developer.android.com/reference/androidx/camera/camera2/interop/CaptureRequestOptions.Builder#clearCaptureRequestOption(android.hardware.camera2.CaptureRequest.Key%3CValueT%3E)), which I wrap for `CaptureRequestOptions` has the same effect, so that's why I didn't worry about it for now.